reward/stakingManager_test.go

```diff
@@ -0,0 +1,43 @@
+package reward
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+// checking calculate blockNumber of stakingInfo and return the stakingInfo with the blockNumber correct when stakingInfo is stored in cache
+func TestStakingManager_getStakingInfoFromStakingCache(t *testing.T) {
+	testData := []uint64{
+		0, 86400, 172800, 259200,
+	}
+	testCases := []struct {
+		stakingNumber  uint64
+		expectedNumber uint64
+	}{
+		{1, 0},
+		{100, 0},
+		{86400, 0},
+		{86401, 0},
+		{172800, 0},
+		{172801, 86400},
+		{200000, 86400},
+		{259200, 86400},
+		{259201, 172800},
+		{300000, 172800},
+		{345600, 172800},
+		{345601, 259200},
+		{400000, 259200},
```

Comment:
Could you please explain how this stackingNumber and expectedNumber pairs have been made?

Reply:
This is the function which calculates the stakingBlockNumber for a given blockNumber
```
func CalcStakingBlockNumber(blockNum uint64) uint64 {
	stakingInterval := StakingUpdateInterval()
	if blockNum <= 2*stakingInterval {
		// Just return genesis block number.
		return 0
	}

	var number uint64
	if (blockNum % stakingInterval) == 0 {
		number = blockNum - 2*stakingInterval
	} else {
		number = blockNum - stakingInterval - (blockNum % stakingInterval)
	}
	return number
}
```